wkcuber/image_readers.py

```diff
@@ -199,11 +208,128 @@ def read_dtype(self, file_name: str) -> str:
             ].dtype.name
 
 
+class CziImageReader(ImageReader):
+    def __init__(self) -> None:
+        self.tile_shape = None
+
+    # returns format (X, Y)
+    def _read_array_single_channel(
+        self, czi_file: CziFile, channel_slice: int, dtype: np.dtype, z_slice: int
+    ) -> np.ndarray:
+        channel_index = czi_file.axes.find("C")
+        x_index = czi_file.axes.find("X")
+        y_index = czi_file.axes.find("Y")
+        z_index = czi_file.axes.find("Z")
+        channel_file_start = czi_file.start[channel_index]
+        z_file_start = czi_file.start[z_index]
+        for entry in czi_file.filtered_subblock_directory:
+            if (
+                entry.start[z_index] - z_file_start == z_slice
+                and entry.start[channel_index] - channel_file_start == channel_slice
+            ):
+                data = entry.data_segment().data()
+                if x_index > y_index:
+                    data = data.reshape(data.shape[y_index], data.shape[x_index])
+                    data = data.swapaxes(0, 1)
+                else:
+                    data = data.reshape(data.shape[x_index], data.shape[y_index])
+                data = to_target_datatype(data, dtype)
+                return data
+
+    # return format will be (X, Y, channel_count)
+    def _read_array_all_channels(
+        self, czi_file: CziFile, dtype: np.dtype, z_slice: int
+    ) -> np.ndarray:
+        x_index = czi_file.axes.find("X")
+        y_index = czi_file.axes.find("Y")
+        z_index = czi_file.axes.find("Z")
+        c_index = czi_file.axes.find("C")
+        indices = [(x_index, "X"), (y_index, "Y"), (c_index, "C")]
+        indices.sort()
```

Comment:
The example file I used had the following axes: `VIHRSCTZYX0` with the following dimensions `(1, 1, 1, 1, 1, 2, 1, 624, 1920, 1920, 1)`. Therefore, I am not sure, whether there is a fix ordering or if all axes are always present.
To make the code work for all cases, I use the sorted indices to reshape this multi-dimensional data into 3 dimensional data. In this example, this would lead to `reshape(2, 1920,1920)` for each tile, which represents a single Z layer. Afterwards, I can use the sorted indices to transpose the data into our desired format of `X,Y,C`. 

I hope my comments are enough to explain this in the code, too.